src/shims/io.rs

```diff
@@ -0,0 +1,179 @@
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+
+use crate::stacked_borrows::Tag;
+use crate::*;
+
+#[derive(Default)]
+pub struct FileHandler {
+    files: HashMap<i32, File>,
+    flags: HashMap<i32, i32>,
+    low: i32,
+}
+
+impl<'mir, 'tcx> EvalContextExt<'mir, 'tcx> for crate::MiriEvalContext<'mir, 'tcx> {}
+pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx> {
+    fn open(
+        &mut self,
+        path_op: OpTy<'tcx, Tag>,
+        flag_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        if !this.machine.communicate {
+            throw_unsup_format!("`open` not available when isolation is enabled")
+        }
+
+        let flag = this.read_scalar(flag_op)?.to_i32()?;
+
+        if flag
+            != this
+                .eval_path_scalar(&["libc", "O_RDONLY"])?
+                .unwrap()
+                .to_i32()?
```

Comment:
I was thinking about writing a `FileHandler::init()` function to populate a dictionary with all the `libc` constants used here so we dont have to do this everywhere. Opinions?

Reply:
idk, If there's enough of these to warrant special code I'd rather have a `eval_libc_i32` function or sth.

Reply:
Ok I'll add it to `src/helpers.rs`